x509/src/general_name.rs

```diff
@@ -0,0 +1,417 @@
+/// GeneralNames as defined in [RFC 5280 Section 4.2.1.6].
+use crate::alloc::string::ToString;
+use crate::Name;
+use der::asn1::{Any, ContextSpecific, Ia5String, ObjectIdentifier, OctetString};
+use der::{Decodable, DecodeValue, Decoder, Length, Sequence, TagMode, TagNumber};
+
+/// OtherName as defined in [RFC 5280 Section 4.2.1.6] in support of the Subject Alternative Name extension.
+///
+/// ```text
+///    OtherName ::= SEQUENCE {
+///         type-id    OBJECT IDENTIFIER,
+///         value      [0] EXPLICIT ANY DEFINED BY type-id }
+/// ```
+///
+/// [RFC 5280 Section 4.2.1.6]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct OtherName<'a> {
+    /// type-id    OBJECT IDENTIFIER,
+    pub type_id: ObjectIdentifier,
+
+    /// value      [0] EXPLICIT ANY DEFINED BY type-id }
+    pub value: Any<'a>,
+}
+
+impl<'a> DecodeValue<'a> for OtherName<'a> {
+    fn decode_value(decoder: &mut Decoder<'a>, _length: Length) -> der::Result<Self> {
+        let type_id = decoder.decode()?;
+        let value = decoder.decode()?;
+        Ok(Self { type_id, value })
+    }
+}
+
+impl<'a> Sequence<'a> for OtherName<'a> {
+    fn fields<F, T>(&self, f: F) -> ::der::Result<T>
+    where
+        F: FnOnce(&[&dyn der::Encodable]) -> ::der::Result<T>,
+    {
+        f(&[&self.type_id, &self.value])
+    }
+}
```

Comment:
Was there an issue deriving `Sequence` on this type?

Reply:
Yes. If I comment out the above and add Sequence back to the derive, the following results:
```
the trait `DecodeValue<'_>` is not implemented for `OtherName<'_>`
```

Reply:
Weird. Guess I'll have to take a look at the `cargo expand` output and see why the derived implementation differs, but I can do that after merging this.